src/plugins/discover/public/application/view_components/canvas/top_nav.tsx

```diff
@@ -65,15 +65,10 @@ export const TopNav = ({ opts, showSaveQuery, isEnhancementsEnabled }: TopNavPro
     ? getTopNavLinks(services, inspectorAdapters, savedSearch, isEnhancementsEnabled)
     : [];
 
-  connectStorageToQueryState(
-    services.data.query,
-    osdUrlStateStorage,
-    {
-      filters: opensearchFilters.FilterStateStore.APP_STATE,
-      query: true,
-    },
-    uiSettings
-  );
+  useConnectStorageToQueryState(services.data.query, osdUrlStateStorage, {
```

Comment:
Since `syncConfig` is a dependency in  `useConnectStorageToQueryState`, this object will regenerate every render. They will unsubscribe  and subscribe again every render. Is this by design? Shall we need to add a useMemo depend on config value directly?